### ISSUE TITLE
cb_explore and cb_explore_adf are taking invalide epsilon arguments

### ISSUE THREAD

**Issue by adityauser** [VowpalWabbit/vowpal_wabbit#2322]
#### Describe the bug
The command `vw -d train_adf.dat --cb_explore_adf  --epsilon -1` is running flawlessly but clearly it is invalid since the epsilon is beyond its bound 

#### To Reproduce
Run any of these commands:
`vw -d train_adf.dat --cb_explore_adf  --epsilon -1`
`vw -d train_adf.dat --cb_explore_adf  --epsilon 2`
`vw -d train.dat --cb_explore 4 --epsilon 4`
`vw -d train_adf.dat --cb_explore_adf --first 2 --epsilon -4`

#### Expected behaviour
Throw an invalid argument error.

#### Environment

`VW 8.8.0 on Linux with the Python bindings`

#### Status

I'm currently working on this bug

**Comment by jackgerrits** [comment]
Yes, I agree epsilon should be between 0 and 1 inclusive.

**Comment by mritunjaysharma394** [comment]
I have tried to fix this bug by prompting an error :
`ERROR: Invalid argument error, epsilon should be between 0 and 1 (both inclusive).`
and it is working for all the cases, except 
`vw -d train.dat --cb_explore 4 --epsilon 4`.
Maybe @jackgerrits you can help me out in cracking this case. Also, do we need to terminate the entire execution or just prompt an error?

**Comment by jackgerrits** [comment]
Is `vw -d train.dat --cb_explore 4 --epsilon 4` valid input or invalid?

An invalid epsilon is fatal.

**Comment by mritunjaysharma394** [comment]
In the fix, I tried to implement, I am producing this kind of output for all other invalid epsilon cases:

![Screenshot from 2020-03-06 00-46-53](https://user-images.githubusercontent.com/45476902/76017183-3338d980-5f44-11ea-9c0f-a3ae8b4db118.png)

Is this the right way? Do we have to use abort() also?
Also, I don't think `vw -d train.dat --cb_explore 4 --epsilon 4` is a valid input. Maybe @adityauser can clarify more on that?

**Comment by mritunjaysharma394** [comment]
After following the suggested instructions, output for respective wrong entries of epsilon is something like this:

```
vw -d train.txt --cb_explore_adf --epsilon 2
Error: cb_explore_adf_greedy_epsilon must be in [0,1]

finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0
cb_explore_adf_greedy_epsilon must be in [0,1]
vw (cb_explore_adf_greedy.cc:125): cb_explore_adf_greedy_epsilon must be in [0,1]
```
Is it fine? Can I send a pull request for the issue?

**Comment by thealphadollar** [comment]
@jackgerrits I guess this has been fixed via #2325 and can be closed safely.
